nixos/modules/services/monitoring/prometheus/exporters/blackbox.nix

```diff
@@ -4,6 +4,13 @@ with lib;
 
 let
   cfg = config.services.prometheus.exporters.blackbox;
+
+  checkConfig = file: pkgs.runCommand "checked-blackbox-exporter.conf" {
+    preferLocalBuild = true;
+    buildInputs = [ pkgs.prometheus-blackbox-exporter ]; } ''
```

Comment:
Although we have `preferLocalBuild` here I guess that this may break in some cross-compilation environments: `pkgs.` references the package set for the target machine (see `pkgs/top-level/stage.nix` for further information), so when deploying an aarch64 target from a local x86_64 build host, you'd try to validate the exporter's config with a tool compiled against aarch64.

The easiest way to work around this here is to use `pkgs.buildPackages`.

Reply:
Thank's for clearing that up, I never really noticed this.
I saw this usage of `runCommand` with `preferLocalBuild` in a couple of other modules and got my copy-pasta from [here](https://github.com/NixOS/nixpkgs/blob/99930cadb46c5256a90dd50a050a3f40f1c35663/nixos/modules/services/monitoring/prometheus/default.nix#L43-L46).